src/Agent.Listener/Agent.cs

```diff
@@ -132,8 +151,8 @@ public override void Initialize(IHostContext hostContext)
                     {
                         try
                         {
-                            HostContext.CancellationToken.ThrowIfCancellationRequested();
-                            message = await listener.GetNextMessageAsync(); //get next message
+                            token.ThrowIfCancellationRequested();
```

Comment:
Currently exceptions, including OperationCanceledException, are passed to the caller and handled i Program.MainAsync. Even if I add While(!token.IsCancellationRequested){}., then istener.GetNextMessageAsync(token) will throw OperationCanceledException, because this is the method where most of the time is spent waiting for the next message.